File: app/src/main/java/workshop1024/com/xproject/controller/adapter/MinimalAdapter.java
package workshop1024.com.xproject.controller.adapter;

import android.content.Context;
import android.databinding.DataBindingUtil;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.List;

import workshop1024.com.xproject.R;
import workshop1024.com.xproject.controller.activity.home.NewsDetailActivity;
import workshop1024.com.xproject.databinding.NewslistItemMinimalBinding;
import workshop1024.com.xproject.model.news.News;

/**
 * Created by chengxiang.peng on 2018/1/24.
 */
public class MinimalAdapter extends RecyclerView.Adapter<MinimalAdapter.MinimalViewHolder> {
    private Context mContext;
    private List<? extends News> mNewsList;

    public MinimalAdapter(Context context, List<? extends News> newsList) {
        mContext = context;
        mNewsList = newsList;
    }

    @Override
    public MinimalViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        NewslistItemMinimalBinding newslistItemMinimalBinding = DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()),
                R.layout.newslist_item_minimal, parent, false);
        newslistItemMinimalBinding.setMinimalHandlers(new MinimalHandlers());
        return new MinimalViewHolder(newslistItemMinimalBinding);
    }

    @Override
    public void onBindViewHolder(MinimalViewHolder holder, int position) {
        holder.mNewslistItemMinimalBinding.setNews(mNewsList.get(position));
    }

    @Override
    public int getItemCount() {
        return mNewsList.size();
    }

    public class MinimalViewHolder extends RecyclerView.ViewHolder {
        private final NewslistItemMinimalBinding mNewslistItemMinimalBinding;

        private MinimalViewHolder(NewslistItemMinimalBinding newslistItemMinimalBinding) {
            super(newslistItemMinimalBinding.getRoot());
            mNewslistItemMinimalBinding = newslistItemMinimalBinding;
        }
    }

    public class MinimalHandlers {
        public void onClickItem(View view, News news) {
            NewsDetailActivity.startActivity(mContext, news.getNewId());
        }
    }
}

File: app/src/main/java/workshop1024/com/xproject/controller/adapter/MinimalAdapter.kt
package workshop1024.com.xproject.controller.adapter

import android.content.Context
import android.databinding.DataBindingUtil
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import workshop1024.com.xproject.R
import workshop1024.com.xproject.controller.activity.home.NewsDetailActivity
import workshop1024.com.xproject.databinding.NewslistItemMinimalBinding
import workshop1024.com.xproject.model.news.News

/**
 * Created by chengxiang.peng on 2018/1/24.
 */
class MinimalAdapter(private val mContext: Context, private val mNewsList: List<News>) : RecyclerView.Adapter<MinimalAdapter.MinimalViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MinimalViewHolder {
        val newslistItemMinimalBinding = DataBindingUtil.inflate<NewslistItemMinimalBinding>(LayoutInflater.from(parent.context),
                R.layout.newslist_item_minimal, parent, false)
        newslistItemMinimalBinding.minimalHandlers = MinimalHandlers()
        return MinimalViewHolder(newslistItemMinimalBinding)
    }

    override fun onBindViewHolder(holder: MinimalViewHolder, position: Int) {
        holder.mNewslistItemMinimalBinding.news = mNewsList[position]
    }

    override fun getItemCount(): Int {
        return mNewsList.size
    }

    inner class MinimalViewHolder(internal val mNewslistItemMinimalBinding: NewslistItemMinimalBinding) : RecyclerView.ViewHolder(mNewslistItemMinimalBinding.root)

    inner class MinimalHandlers {
        fun onClickItem(view: View, news: News) {
            NewsDetailActivity.startActivity(mContext, news.newId)
        }
    }
}
